daemon/graphdriver/driver.go

```diff
@@ -142,10 +142,36 @@ func GetDriver(name, home string, options []string) (Driver, error) {
 func New(root string, options []string) (driver Driver, err error) {
 	for _, name := range []string{os.Getenv("DOCKER_DRIVER"), DefaultDriver} {
 		if name != "" {
+			logrus.Infof("[graphdriver] trying provided driver %q", name) // so the logs show specified driver
 			return GetDriver(name, root, options)
 		}
 	}
 
+	// Guess for prior driver
```

Comment:
The fact that this is just guessing based on in-directory state is a little scary IMO (which is why what I described in https://github.com/docker/docker/pull/11962#issuecomment-88274858 included bits about explicitly recording requested and acheived states).  Using this as a hint for "auto" in the case of not having any explicit state for which should be selected seems sane as a transition mechanism, but maybe I'm just being over-paranoid?